includes/providers/gam/class-gam-scripts.php

```diff
@@ -255,80 +210,216 @@ function( $a, $b ) {
+						 *
+						 * @see https://developers.google.com/doubleclick-gpt/guides/ad-sizes#responsive_ads
+						 */
+						?>
+						var mapping = googletag.sizeMapping();
+						<?php
+						// Default base is to not show ads.
+						?>
+						var baseSizes = [];
+						<?php
+						// If the ad unit is fluid, base includes fluid.
+						?>
+						if( ad_unit['fluid'] ) {
+							baseSizes = baseSizes.concat( 'fluid' );
+						}
+						<?php
+						// Apply size map rules.
+						?>
+						for ( viewportWidth in ad_unit['size_map'] ) {
+							var width = parseInt( viewportWidth );
+							var mappedSizes = ad_unit['size_map'][ viewportWidth ];
+							mapping.addSize( [ width, 0 ], baseSizes.concat( mappedSizes ) );
+						}
+						<?php
+						// Sticky ads should only be shown on mobile (screen width <=600px).
+						?>
+						if ( ad_unit['sticky'] ) {
+							mapping.addSize( [600, 0], baseSizes );
+						}
+						<?php
+						// On viewports smaller than the smallest ad size, don't show any ads.
+						?>
+						mapping.addSize( [0, 0], baseSizes );
+						defined_ad_units[ container_id ].defineSizeMapping( mapping.build() );
+					}
+
+					if ( ad_config['disable_initial_load'] ) {
+						googletag.pubads().disableInitialLoad();
+					}
+					googletag.pubads().collapseEmptyDivs();
+					googletag.pubads().enableSingleRequest();
+					if ( lazy_load && lazy_load.active ) {
+						googletag.pubads().enableLazyLoad( {
+							fetchMarginPercent: lazy_load.fetch_margin_percent,
+							renderMarginPercent: lazy_load.render_margin_percent,
+							mobileScaling: lazy_load.mobile_scaling
+						} );
+					}
+					googletag.enableServices();
+
+					for ( var container_id in defined_ad_units ) {
+						googletag.display( container_id );
+					}
 
-				<?php
-				/**
-				 * Identify fluid rendered ad and fix iframe width.
-				 * GPT currently sets the iframe with `min-width` set to 100% and property `width` set to 0.
-				 * This causes the iframe to be rendered with 0 width.
-				 */
-				?>
-				googletag.pubads().addEventListener( 'slotRenderEnded', function(event) {
-					var sizes = event.slot.getSizes();
-					if (
-						( event.size === null || event.size[0] === 0 ) &&
-						Array.isArray( sizes ) && sizes.indexOf( 'fluid' ) !== -1
-					) {
+					googletag.pubads().addEventListener( 'slotRenderEnded', function( event ) {
 						var container = document.getElementById( event.slot.getSlotElementId() );
-						if ( container ) {
-							var iframe = container.querySelector( 'iframe' );
-							if ( iframe ) {
-								iframe.style.width = '100%';
+						if ( ! container ) {
+							return;
+						}
+						var ad_unit = container.ad_unit;
+						if ( ! ad_unit ) {
+							return;
+						}
+						<?php
+						/**
+						 * Lock fixed height ad unit size mapping to prevent larger
+						 * creatives to render on refresh control.
+						 */
+						?>
+						if ( ad_unit.fixed_height && container.parentNode.style.height === 'auto' ) {
```

Comment:
In a particular setup on a local site, if the impression was not filled, the `size` of the `event` was `null`, which caused an error to be thrown. There should be a check for `event`'s `size`:

```suggestion
						if ( ad_unit.fixed_height && container.parentNode.style.height === 'auto' && event.size ) {
```